File: Comp/three/three.java
import java.util.*;

public class three{
  public static void main(String[] args) {
    Scanner in = new Scanner(System.in);

    // Read the number of test cases
    int T = in.nextInt();
    in.nextLine();

    // Process each test case
    for (int t = 0; t < T; t++) {
      // Read N and M
      int N = in.nextInt();
      int M = in.nextInt();
      in.nextLine();

      // Create a map to store the input-output pairs
      Map<String, Integer> pairs = new HashMap<>();

      // Read the input-output pairs
      for (int m = 0; m < M; m++) {
        String input = in.nextLine();
        int output = Character.getNumericValue(input.charAt(N));
        pairs.put(input.substring(0, N), output);
      }

      // Assume that Elsie is telling the truth
      boolean isLie = false;
      ArrayList<Character> digits = new ArrayList<Character>(); 

      // Check if there is any input-output pair that cannot be explained by the program
      for (Map.Entry<String, Integer> entry : pairs.entrySet()) {
        String input = entry.getKey();
        int output = entry.getValue();
        for(int i=0;i<input.length();i++){
            digits.add(input.charAt(i));
        }
      }
      // Print the result for this test case
      if (isLie) {
        System.out.println("LIE");
      } else {
        System.out.println("OK");
      }
    }
  }
}

File: Comp/three/three.py
def elsie_must_be_lying(t, test_cases):
  for i in range(t):
    n, m, inputs_outputs = test_cases[i]
    # Initialize a set to store the outputs for each input
    output_set = set()
    for input_output in inputs_outputs:
      b, output = input_output
      # If the input has not been seen before, add it to the set
      # with its output. If it has been seen before, check that
      # the output is the same as the one previously seen for
      # this input. If it is different, Elsie must be lying.
      if b not in output_set:
        output_set[b] = output
      else:
        if output_set[b] != output:
          print("LIE")
          break
    # If we have not found any inconsistencies, the output is "OK"
    else:
      print("OK")

# Test the function with the sample input
elsie_must_be_lying(4, [
  (1, 3, [("0", 0), ("0", 0), ("1", 1)]),
  (2, 4, [("00", 0), ("01", 1), ("10", 1), ("11", 1)]),
  (1, 2, [("0", 1), ("0", 0)]),
  (2, 4, [("00", 0), ("01", 1), ("10", 1), ("11", 0)]),
])
